cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java

```diff
@@ -125,8 +129,9 @@ public Integer doAction() throws Exception {
    * Resolves dependencies from templates
    *
    * @throws IOException
+   * @throws UpgradeTemplatesNotificationException
    */
-  private void resolveTemplateDependencies() throws IOException {
+  private void resolveTemplateDependencies() throws IOException, UpgradeTemplatesNotificationException {
```

Comment:
I am not sure, why I cant find the exception  UpgradeTemplatesNotificationException as part of this PR. Has it been integrated beforehand in another PR? 
Is this Exception inheriting the Cobigen creation exception?